pkg/mdformatter/linktransformer/validator.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package linktransformer
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	gitHubAPIURL = "https://api.github.com/repos/%v/%v?sort=created&direction=desc&per_page=1"
+)
+
+type GitHubResponse struct {
+	Number int `json:"number"`
+}
+
+type URLValidator struct {
+	Matched bool
+}
+
+// GetValidatorForURL matches link with any one of provided validators.
+func (v Config) GetValidatorForURL(url string) URLValidator {
```

Comment:
We need to work on different abstrations here...

What about something else:

```
type Validator interface {
IsValid(URL string) (bool, error)
}
```

* Type == github, where we can have githubValidator struct, which does gh check
* Type  == rt rtValidator, which does normal look up 
![image](https://user-images.githubusercontent.com/6950331/120312628-e2dc3300-c2d0-11eb-9d53-25c4a670d3f4.png)
* Type == static/ignore which is always true?